src/main/scala/strawman/collection/Factories.scala

```diff
@@ -1,133 +1,165 @@
 package strawman
 package collection
 
-import strawman.collection.mutable.Builder
+import scala.language.implicitConversions
+
+import strawman.collection.mutable.{ArrayBuffer, Builder}
 
 import scala.{Any, Int, Nothing, Ordering}
 import scala.annotation.unchecked.uncheckedVariance
 
+
+/** Builds a collection of type `C` from elements of type `A` when a source collection of type `From` is available.
+  * Implicit instances of `BuildFrom` are available for all collection types.
+  *
+  * @tparam From Type of source collection
+  * @tparam A Type of elements (e.g. `Int`, `Boolean`, etc.)
+  * @tparam C Type of collection (e.g. `List[Int]`, `TreeMap[Int, String]`, etc.)
+  */
+trait BuildFrom[-From, -A, +C] extends Any {
```

Comment:
I would prefer a type member (and I did use one in my original implementation in #45) but `FromSpecificIterable` in Martin's simplified version uses a type parameter so I used the same here. I think we should use type members for both. Unlike `CanBuildFrom` lookup for `BuildFrom` and `FromSpecificIterable` is never determined by this type. It is strictly an "out" parameter.